core/src/test/java/io/questdb/test/cutlass/http/ScoreboardHandlingTest.java

```diff
@@ -0,0 +1,174 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.cutlass.http;
+
+import io.questdb.PropertyKey;
+import io.questdb.ServerMain;
+import io.questdb.std.Files;
+import io.questdb.std.str.Path;
+import io.questdb.test.AbstractBootstrapTest;
+import io.questdb.test.tools.TestUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static io.questdb.test.tools.TestUtils.*;
+
+public class ScoreboardHandlingTest extends AbstractBootstrapTest {
+
+    static String backupRoot;
+
+    @BeforeClass
+    public static void setUpStatic() throws Exception {
+        AbstractBootstrapTest.setUpStatic();
+        unchecked(() -> {
+            backupRoot = temp.newFolder("backups").getAbsolutePath();
+        });
+    }
+
+    @Test
+    public void testPermissionsAreUpdatedWhenColumnIsRenamed() throws Exception {
+
+        createDummyConfiguration();
+        assertMemoryLeak(() -> {
+            try (
+                    ServerMain main = startWithEnvVariables(
+                            PropertyKey.DEV_MODE_ENABLED.getEnvVarName(), "true",
+                            PropertyKey.CAIRO_SQL_BACKUP_ROOT.getEnvVarName(), backupRoot
+
+                    );
+                    TestHttpClient httpClient = new TestHttpClient()
+            ) {
+                TestUtils.executeSQLViaPostgres(
+                        main.getConfiguration().getPGWireConfiguration().getDefaultUsername(),
+                        main.getConfiguration().getPGWireConfiguration().getDefaultPassword(),
+                        main.getPgWireServerPort(),
+                        "create table trades(ts timestamp, ticker symbol index, value double) timestamp(ts) partition by day wal"
+                );
+
+                exec(
+                        httpClient,
+                        main,
+                        "{\"query\":\"select ticker from trades\",\"columns\":[{\"name\":\"ticker\",\"type\":\"SYMBOL\"}],\"timestamp\":-1,\"dataset\":[],\"count\":0}",
+                        "select ticker from trades"
+                );
+
+                dml(httpClient, main, "insert into trades values ('2023-07-01T00:00:00.000000', 'ABC', 1.0)");
+                dml(httpClient, main, "insert into trades values ('2023-07-02T00:00:00.000000', 'DEF', 2.0)");
+
+                // this is a bad naming schema: "updated" is a txn, not the number of updated rows
```

Comment:
The confusion seems to come from WAL/non-WAL: `UpdateOperation#apply()` indeed returns the result of `UpdateOperatorImpl#executeUpdate()` which in its turn returns the number of updated rows.